api/s2n.h

```diff
@@ -15,12 +15,25 @@
 
 #pragma once
 
+#define S2N_GCC_VERSION (__GNUC__ * 10000 \
```

Comment:
Why is this in api.h? I would argue against this as it will pollute the application's scope.

Reply:
I put this here so we can have a mechanism to correctly deprecate API calls. This was originally in s2n_compiler.h, which wasn't available when being built in some internal dependencies.

We could have a one-off method of detecting GCC versions < 4.5, but it will be hard to read and not standard across the code base. Having the version detection in s2n.h allows compiler attributes to be used everywhere.

Reply:
I think to unblock getting the fix out, it would be better to have this in s2n_compiler.h. there aren't any current needs for the deprecation attribute other than the enable TLS 1.3 call. We can later discuss if this should be moved to a more accessible location.